### ISSUE TITLE
Properties in the generic step extension are not grouped in the correct way

### ISSUE THREAD

**Issue by lhein** [KaotoIO/kaoto-backend#474]
### Describe the Bug

The Camel Catalog provides information for every property of a Camel Component which group that property belongs to. Currently we do not group in the correct way (looks quite random).

![grafik](https://user-images.githubusercontent.com/435853/219472124-5cd01220-b0d6-4da4-b7bb-e16e5f6d624e.png)

The original Camel catalog should be leveraged to group the attributes correctly. I would suggest to use collapsible sections for each group to make it easier for users to filter what they are after.

In the Eclipse Camel Editor it was done like this:
![grafik](https://user-images.githubusercontent.com/435853/219472710-4036000f-88be-486f-a170-31b92ebd1df8.png)

### Steps to Reproduce the Bug or Issue

1. select Camel Route type
2. select IMAP as source
3. select the IMAP node on the canvas to open the configuration panel

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 0.30.0]

**Comment by lhein** [comment]
Where do section names in the above screenshot come from?

**Comment by Delawen** [comment]
Those are not section names, those are properties that are of type object and therefore the default config tab does not know how to configure them.

**Comment by lhein** [comment]
If we cannot configure them, why bother users displaying them?

**Comment by kahboom** [comment]
>Should we add some kind of category property to the parameters attributes too? And if there is no category defined, do I left it empty or do we agree on a default category?

I don't mind the idea of having some category for grouping properties, and maybe expanding/collapsing them for the default view. And if it's only one category or none we can just default to show them as we do now.

>If we cannot configure them, why bother users displaying them?

@lhein maybe @Delawen has a better idea here, but I think it's because it'd be too much work to remove those properties, when instead we could just have step extensions to better handle supporting them, as sometimes they can be complex objects (we just don't know and it depends on each step).

It'd be pretty impossible to handle any complex object without a more formal schema for Kamelets (which we don't have). We can maybe add better default support in the UI for handling simple objects.

**Comment by lhein** [comment]
Right....that's why I think it is not worth displaying these properties in the generic form if we can't change them. For custom step extensions this is handled differently anyways. If the backend has no idea if the frontend can handle it then I agree that filtering would be too much work. In that case I would change the way of visualizing them. Currently they look like headline and are displayed bold. That is an eye catcher and easier to find than mandatory field marked with just a tiny *. Wdyt?

**Comment by kahboom** [comment]
>Right....that's why I think it is not worth displaying these properties in the generic form if we can't change them. For custom step extensions this is handled differently anyways. If the backend has no idea if the frontend can handle it then I agree that filtering would be too much work. In that case I would change the way of visualizing them. Currently they look like headline and are displayed bold. That is an eye catcher and easier to find than mandatory field marked with just a tiny *. Wdyt?

That's correct, the backend has no idea if the frontend can handle it. I think (if we can) we should format the fields better, but I would need to study the majority of the objects that are returned, to see if we can handle a majority of the use cases with some custom fields for the objects. And maybe by grouping them with categories, as per Maria's suggestion. These would be enhancements that I'm not 100% sure would cover most use cases, but we can give it a shot.

**Comment by lhein** [comment]
![grafik](https://user-images.githubusercontent.com/435853/226738878-28e3ecb9-ad74-4b13-83bf-09a500d5a003.png)

**Comment by lhein** [comment]
The grouping information above is available for Camel Components only, but usually these have a lot more parameters in comparison to EIPs, Languages and Data Formats. We should show mandatory fields on top and then create collapsable sections for each of the groups and put in the corresponding parameters in alphabetic order as mentioned by Richard.

**Comment by lhein** [comment]
buzz off, stale bot! ;)
